File: exs/mundo_2/python/037.py
"""
Desafio 037

Problema: Escreva um programa em Python que leia um número
          inteiro qualquer e peça para o usuário escolher qual
          será a base de conversão: 1 para binário, 2 para octal e 3 para hexadecimal.

Resolução do problema:
"""
numero = int(input('Informe um número: '))
print('=-' * 10)
print('[ 1 ] BINÁRIO \n[ 2 ] OCTAL \n[ 3 ] HEXADECIMAL')
print('=-' * 10)

op = int(input('Escolha uma opção: '))
if op == 1:
    print('O valor {} convertido para BINÁRIO É: {}.'.format(numero, bin(numero)[2:]))
elif op == 2:
    print('O valor {} convertido para OCTAL É: {}.'.format(numero, oct(numero)[2:]))
elif op == 3:
    print('O valor {} convertido para HEXADECIMAL É: {}.'.format(numero, hex(numero)[2:]))
else:
    print('Opção INVÁLIDA.')

File: exs/mundo_2/java/037.java
/*
Desafio 037

Problema: Escreva um programa em Python que leia um número
          inteiro qualquer e peça para o usuário escolher qual
          será a base de conversão: 1 para binário, 2 para octal e 3 para hexadecimal.

Resolução do problema:
*/

import java.util.Scanner;

class Main{
    public static void main(String[] args) {

        // instancia a classe Scanner
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite um número: ");
        int numero = scanner.nextInt();

        System.out.printf("Você digitou %d", numero);
        System.out.println("\n\nEscolha uma opção abaixo: \n[ 1 ] CONVERSÃO PARA BINÁRIO\n[ 2 ] CONVERSÃO PARA OCTAL\n[ 3 ] CONVERSÃO PARA HEXADECIMAL");
        int op = scanner.nextInt();

        if (op == 1) {
            System.out.printf("O valor %d convertido para BINÁRIO É: %s.%n", numero, Integer.toBinaryString(numero));
        } else if (op == 2) {
            System.out.printf("O valor %d convertido para OCTAL É: %s.%n", numero, Integer.toOctalString(numero));
        } else if (op == 3) {
            System.out.printf("O valor %d convertido para HEXADECIMAL É: %s.%n", numero, Integer.toHexString(numero).toUpperCase());
        } else {
            System.out.println("Opção INVÁLIDA.");
        }

        scanner.close();

    }
}